### ISSUE TITLE
predict.glmmTMB doesn't work if response is logical

### ISSUE THREAD

**Issue by mebrooks** [glmmTMB/glmmTMB#178]
This is not important because there's an easy work around. I just want to document it. In the example below, `mod1` and `mod2` appear to be the same, but `predict.glmmTMB` only works on `mod1`.
```
load(system.file("testdata","radinger_dat.RData",package="lme4"))
mod1 <- glmmTMB(presabs~predictor+(1|species),family=binomial,
                 radinger_dat)
                 
mod2 <- glmmTMB(as.logical(presabs)~predictor+(1|species),family=binomial,
                 radinger_dat)
                 
summary(mod1)
summary(mod2)

predict(mod1)
predict(mod2)
```
It gives the error
```
> predict(mod2)
Problem with these data entries:
yobs 
   6 
Error in MakeADFun(data.tmb, parameters, random = randomArg, profile = NULL,  : 
  Only numeric matrices, vectors, arrays, factors or lists can be interfaced
```
The `6` in the error is consistent across the examples I've tried.

**Comment by bbolker** [comment]
I have to say that one of the things I was sort of hoping to leave behind with `glmmTMB` was the profusion of "oh, you can specify your binary responses any old way you want".  I know we don't allow two-column matrices of responses, forcing people to specify values + weights; have you experimented with whether we allow factors as response variables/do something sensible if users provide them?  (It always struck me as a little crazy that you could specify a factor with *more than two levels* as a response, and `glm` would happily assume that the first level was 'failure' and all other levels were 'success'. I would guess that 99% of the time someone providing a 3+-level factor to `glm` is confused or making a mistake.) From `?family`, under `Details:`

> 1. As a factor: ‘success’ is interpreted as the factor not having the first level (and hence usually of having the  second level).

We could coerce `yobs` to numeric at line 56 of  `glmmTMB.R`, but I'm not sure that's the right solution.  `binomial()$initialize` seems to be where R keeps all of the machinery that does this stuff, but it doesn't enforce the logical-to-numeric conversion (probably because the coercion happens automatically somewhere else?)

**Comment by bbolker** [comment]
closed by c75f2a6
